packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -483,7 +493,10 @@ class PubspecYaml {
     assert(versions != null);
     final List<String> output = <String>[]; // the string data to output to the file, line by line
     final Set<String> directDependencies = new Set<String>(); // packages this pubspec directly depends on (i.e. not transitive)
+    final Set<String> devDependencies = new Set<String>();
     Section section = Section.other; // the section we're currently handling
+
+    int endOfDirectDependencies;
     int lastPossiblePlace; // the line number where we're going to insert the transitive dependencies
```

Comment:
probably worth renaming this one to be consistent with the one you added (also, comment needs updating)